eitprocessing/datahandling/intervaldata.py

```diff
@@ -19,7 +19,7 @@ class Interval(NamedTuple):
 
 
 @dataclass(eq=False)
-class IntervalData(Equivalence, SelectByIndex, HasTimeIndexer):
+class IntervalData(DataContainer, SelectByIndex, HasTimeIndexer):
```

Comment:
In my mind, it would make sense to add `SelectByTime` to `DataContainer` as well. If selecting by time is not possible for `IntervalData` (it should be, though, shouldn't it?*), then you can overwrite the method here to return an error.

*: It's still not clear to me why the `HasTimeIndexer` object is separate from the `TimeIndexer` itself. Also, given that the object inherits both `SelectByIndex` and `HasTimeIndexer`, I think this means that it may as well inherit `SelectByTime`.

Reply:
I don't agree we should have a class inherit functionality to then disable it again. It's possible, but it makes the code less readable. I'll look into it when looking into homogenisation (#279).

If a class inherits from `HasTimeIndexer` it gets an attribute that is an instance of `TimeIndexer`. You can't inherit from a class that needs to be an attribute.